lib/spack/spack/solver/asp.py

```diff
@@ -2028,9 +2028,12 @@ def _spec_clauses(
                     for variant_def in variant_defs:
                         self.variant_values_from_specs.add((spec.name, id(variant_def), value))
 
-                clauses.append(f.variant_value(spec.name, vname, value))
                 if variant.propagate:
                     clauses.append(f.propagate(spec.name, fn.variant_value(vname, value)))
+                    if vname in self.pkg_class(spec.name).variants:
```

Comment:
This is incorrect since #44425 got merged:
```suggestion
                    if self.pkg_class(spec.name).has_variant(vname):
```
Unfortunately fixing this seems to make `test_concretize_propagate_multiple_multivalue_variant` FAIL.